libs/styles/article-card.css

```diff
@@ -0,0 +1,131 @@
+/* article card styles, share between blocks */
+
+.article-card {
+  cursor: pointer;
+  box-sizing: border-box;
+  display: inline-block;
+  /* border: solid 1px var(--color-gray-200); */
+  border-radius: 4px;
+  width: 100%;
+  min-width: 268px;
+  max-width: 378px;
+  margin-bottom: 24px;
+}
+
+@media (min-width: 600px) {
+  .article-card {
+    min-width: unset;
+    width: calc(100% / 2 - 16px);
+    margin-right: 32px;
+    margin-bottom: 32px;
+  }
+  .article-card:nth-of-type(2n),
+  .article-card:last-of-type {
+    margin-right: 0;
+  }
+}
+
+@media (min-width: 1200px) {
+  .article-card {
+    width: calc(100% / 3 - 22px);
+  }
+  .article-card:nth-of-type(2n) {
+    margin-right: 32px;
+  }
+  .article-card:nth-of-type(3n),
+  .article-card:last-of-type {
+    margin-right: 0;
+  }
+}
+
+main a.article-card:any-link {
+  text-decoration: none;
+}
+
+.article-card .article-card-image {
+  width: 100%;
+  height: 250px;
+  line-height: 0;
+  background-color: var(--color-black);
+  border-radius: 4px 4px 0 0;
+}
+
+.article-card .article-card-image img {
+  border-radius: 4px 4px 0 0;
+  object-fit: cover;
+  height: 100%;
+  width: 100%;
+  transition: opacity 0.4s;
+  color: var(--color-white);
+  line-height: var(--body-line-height);
+}
+
+.article-card .article-card-image img:hover {
+  opacity: 0.6;
+}
+
+.article-card .article-card-body {
+  background: var(--color-white);
+  padding: 2rem;
+  border-radius: 0 0 4px 4px;
+  text-align: left;
+}
+
+.article-card .article-card-body p,
+.article-card .article-card-body h3 {
+  margin-left: 0;
+  margin-right: 0;
+  padding-left: 0;
+  padding-right: 0;
+}
+
+.article-card-body p.article-card-category,
+.article-card-body p.article-card-category a,
+.article-card-body p.article-card-date {
+  font-weight: var(--type-detail-all-weight);
+  color: var(--color-gray-600);
+  font-size: var(--type-detail-m-size);
+  line-height: var(--type-detail-m-lh);
+  text-transform: uppercase;
+  letter-spacing: 0.1em;
+  margin-bottom: 0.5rem;
+  text-decoration: none;
+}
+
+.article-card-body h3 {
+  font-size: var(--type-heading-xs-size);
+  line-height: 1.25;
+  margin: 0;
+  margin-bottom: 1rem;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: -webkit-box;
```

Comment:
I'm not familiar with `-webkit-box`. I question using something prefixed because it's non-standard

Reply:
@meganthecoder  `-webkit-box` is non-standard, but major browsers still support it on both desktop and mobile. It is used here for truncating the text. Would it be better to truncate using javascript instead?

Reply:
CSS is definitely preferable to js. If it passing our testing matrix, then I'm cool with it. So basically, lastest Chrome, latest firefox, safari 15 and above, safari 14 most things should work, and Chromium Edge.